File: 898-transpose-matrix/transpose-matrix.py
class Solution:
    def transpose(self, matrix: List[List[int]]) -> List[List[int]]:
        rows=len(matrix)
        cols=len(matrix[0])
        
        res=[]
        for i in range(cols):
            row_list=[]
            for j in range(rows):
                row_list.append(matrix[j][i])
            res.append(row_list)
        return res

File: 898-transpose-matrix/transpose-matrix.java
class Solution {
    public int[][] transpose(int[][] matrix) {
       int m=matrix.length; //rows
       int n=matrix[0].length; //cols
       int[][] res=new int[n][m]; //res[cols][row]

       for(int i=0;i<n;i++){//cols
        for(int j=0;j<m;j++){//rows
            res[i][j]=matrix[j][i];
        }
       }

       return res;
    }
}


/*

 int n=matrix.length;
        for(int i=0;i<n;i++){
            for(int j=0;j<n;j++){
                if(i<j){
                    int temp=matrix[i][j];
                    matrix[i][j]=matrix[j][i];
                    matrix[j][i]=temp;
                }
            }
        }
        return matrix;

*/